### PR TITLE
Inherit annotations from upper resources to k8s podgroup.

### PR DESCRIPTION
**Pull request by jiangkaihua**
When users applied k8s origin workloads like `daemonset`, `job`, etc., k8s created pods first, then volcano created podgroup from the pods, whose progress was different form volcano jobs. So podgroup created from k8s pods would miss annotations of origin workloads, causing configurations inserted in the form of annotations neglected, like #1901 .

So this pr fetched annotations from upper resources by searching pod `ownerReferences`, and filled in podgroups, so that some configurations in annotations were available for k8s workloads.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -27,7 +27,7 @@ require (
 	k8s.io/kubernetes v1.23.0
 	sigs.k8s.io/yaml v1.2.0
 	stathat.com/c/consistent v1.0.0
-	volcano.sh/apis v0.0.0-20211109104455-7e8d893a9f66
+	volcano.sh/apis v1.5.0-beta.0.0.20220112022954-fd6c584c7994
```
Comment (hwdef) [inline go.mod:30]: I'm not sure if this version number is spelled correctly, it seems to be different than in another pr

https://github.com/volcano-sh/volcano/pull/1964/files#diff-33ef32bf6c23acb95f5902d7097b7a1d5128ca061167ec0716715b0b9eeaa5f6R30

Comment (jiangkaihua) [reply go.mod]: Yes, I've updated `apis` vendor to latest master branch.

File: pkg/scheduler/api/job_info.go [hunk 2]
```diff
@@ -365,12 +365,20 @@ func (ji *JobInfo) SetPodGroup(pg *PodGroup) {
...
 	var err error
-	ji.WaitingTime, err = ji.extractWaitingTime(pg)
+	ji.WaitingTime, err = ji.extractWaitingTime(pg, v1beta1.JobWaitingTime)
 	if err != nil {
 		klog.Warningf("Error occurs in parsing waiting time for job <%s/%s>, err: %s.",
 			pg.Namespace, pg.Name, err.Error())
 		ji.WaitingTime = nil
 	}
+	if ji.WaitingTime == nil {
+		ji.WaitingTime, err = ji.extractWaitingTime(pg, JobWaitingTime)
```
Comment (shinytang6) [inline pkg/scheduler/api/job_info.go:375]: why we have different keys `JobWaitingTime` and `v1beta1.JobWaitingTime` here?

Comment (jiangkaihua) [reply pkg/scheduler/api/job_info.go:375]: The old key `JobWaitingTime` is only available to volcano jobs, whereas new key `v1beta1.JobWaitingTime` can be used in both volcano job and k8s workloads. So new key can substitute the old one basically.

The old key was reserved temporarily to ensure that previous users still could use SLA plugin as usual. After all annotation-keys were available, a doc would be proposed to introduce these keys, and then old keys could be removed.

**Comment by Thor-wl** [comment]
Pls rebase from the master branch and solve the conflicts.
@jiangkaihua

**Review by william-wang — COMMENTED** [review]
/lgtm

**Review by william-wang — APPROVED** [review]
/lgtm
/approve
